ios/Video/NowPlayingInfoCenterManager.swift

```diff
@@ -0,0 +1,266 @@
+    private var pauseTarget: Any?
+    private var skipForwardTarget: Any?
+    private var skipBackwardTarget: Any?
+    private var playbackPositionTarget: Any?
+    private var seekTarget: Any?
+
+    private var recivingRemoveControlEvents = false {
+        didSet {
+            recivingRemoveControlEvents ? UIApplication.shared.endReceivingRemoteControlEvents() : UIApplication.shared.beginReceivingRemoteControlEvents()
+        }
+    }
+
+    deinit {
+        cleanup()
+    }
+
+    func registerPlayer(player: AVPlayer) {
+        if players.contains(player) {
+            return
+        }
+
+        if recivingRemoveControlEvents == false {
+            recivingRemoveControlEvents = true
+        }
+
+        if let oldObserver = observers[player.hashValue] {
+            oldObserver.invalidate()
+        }
+
+        observers[player.hashValue] = observePlayers(player: player)
+        players.add(player)
+
+        if currentPlayer == nil {
+            setCurrentPlayer(player: player)
+        }
+    }
+
+    func removePlayer(player: AVPlayer) {
+        if !players.contains(player) {
+            return
+        }
+
+        if let observer = observers[players.hashValue] {
+            observer.invalidate()
+        }
+
+        observers.removeValue(forKey: players.hashValue)
+        players.remove(player)
+
+        if currentPlayer == player {
+            currentPlayer = nil
+            updateMetadata()
+        }
+
+        if players.allObjects.isEmpty {
+            cleanup()
+        }
+    }
+
+    public func cleanup() {
+        observers.removeAll()
+        players.removeAllObjects()
+
+        if let playbackObserver {
+            currentPlayer?.removeTimeObserver(playbackObserver)
+        }
+
+        let commandCenter = MPRemoteCommandCenter.shared()
+        invalideTargets(commandCenter)
+
+        MPNowPlayingInfoCenter.default().nowPlayingInfo = [:]
+        recivingRemoveControlEvents = false
+    }
+
+    private func setCurrentPlayer(player: AVPlayer) {
+        if player == currentPlayer {
+            return
+        }
+
+        if let playbackObserver {
+            currentPlayer?.removeTimeObserver(playbackObserver)
+        }
+
+        currentPlayer = player
+        registerTargets()
+
+        updateMetadata()
+
+        // one secound observer
+        playbackObserver = player.addPeriodicTimeObserver(
+            forInterval: CMTime(value: 1, timescale: 4),
+            queue: .global(),
+            using: { [weak self] _ in
+                self?.updatePlaybackInfo()
+            }
+        )
+    }
+
+    private func registerTargets() {
+        let commandCenter = MPRemoteCommandCenter.shared()
+
+        invalideTargets(commandCenter)
+
+        playTarget = commandCenter.playCommand.addTarget { [weak self] _ in
+            guard let self, let player = self.currentPlayer else {
+                return .commandFailed
+            }
+
+            if player.rate == 0 {
```

Comment:
That is sad to test rate to play ...
I think there is a "global issue" mixing rate and pause/playback state ...

Reply:
According to Apple documentation 
![CleanShot 2024-05-06 at 10 59 27](https://github.com/react-native-video/react-native-video/assets/83216158/97a7896d-797a-4295-9a3f-bc481fec49ec)
So I think that checking `rate == 0` to check if player is paused in good practice